### PR TITLE
feat(activity-log): adds the activity log to the feature flags page

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

Now we're gathering feature flag activity. We need to display it somewhere

## Changes

To minimise the amount of UI needing to change before we can start displaying the activity log. This adds the log to the feature flags page.

New UI is added behind flag 8545-ff-activity-log

See design at https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=5346%3A28877

Still TODO for this PR:
- [ ] activity isn't updating when the loader returns new values?
- [ ] add a cypress test for FF on and off?

Follow up for another PR
- [ ] paging on the API and in the UI

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

added tests, ran it locally

### CODE REVIEW

**Comment by pauldambra** [comment]
loading state

loading-state

**Comment by Twixes** [comment]
Half a cent from me: The loading bar seems a bit redundant when animated skeleton elements already convey the same. The bar is also very attention-grabbing at scene-level like this, so considering the skeleton view, I think the loading experience may be more seamless without it. 🤔

File: frontend/src/lib/components/ActivityLog/ActivityLog.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    <div>
+                        <strong>{logItem.name ?? 'unknown user'}</strong> {logItem.description}
+                    </div>
+                    <div className={'text-muted'}>
+                        <TZLabel time={logItem.created_at} />
+                    </div>
+                </div>
+            </div>
+        )
+    })
```
Comment (mariusandra) [inline frontend/src/lib/components/ActivityLog/ActivityLog.tsx:59]: nit: I don't see a real need for a separate variable for these rows. I think the component will be easier to read if you inline it.

File: frontend/src/scenes/feature-flags/FeatureFlags.tsx [hunk 2]
```diff
@@ -177,11 +182,54 @@ export function FeatureFlags(): JSX.Element {
...
+                            type="primary"
+                            to={urls.featureFlag('new')}
+                            data-attr="new-feature-flag"
+                            icon={<PlusOutlined />}
+                        >
+                            New Feature Flag
+                        </LinkButton>
+                    ) : (
+                        <></>
+                    )
```
Comment (mariusandra) [inline frontend/src/scenes/feature-flags/FeatureFlags.tsx:213]: nit `: null` is less code than the `React.createElement(React.fragment(...))` that happens with `<></>`

File: frontend/src/lib/components/ActivityLog/activityLogLogic.test.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import { expectLogic } from 'kea-test-utils'
+import { dayjs } from 'lib/dayjs'
+import React from 'react'
+import { useMocks } from '~/mocks/jest'
+import { urls } from 'scenes/urls'
+import { Link } from 'lib/components/Link'
+import {
+    ActivityScope,
+    HumanizedActivityLogItem,
+    registerActivityDescriptions,
```
Comment (mariusandra) [inline frontend/src/lib/components/ActivityLog/activityLogLogic.test.tsx:11]: ts import error on `registerActivityDescriptions`

File: frontend/src/lib/components/ActivityLog/humanizeActivity.test.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    detail: {
+        changes: changes,
+        name,
+    },
+    created_at: '2022-02-05T16:28:39.594Z',
+})
+
+describe('humanizing the activity log', () => {
+    describe('humanizing feature flags', () => {
+        registerActivityDescriptions({ scope: ActivityScope.FEATURE_FLAG, describer: flagActivityDescriber })
```
Comment (mariusandra) [inline frontend/src/lib/components/ActivityLog/humanizeActivity.test.tsx:28]: probably not

File: frontend/src/scenes/feature-flags/activityDescriptions.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    }
+    const descriptions = []
+    if (logItem.activity == 'created') {
+        descriptions.push(<>created the flag: {nameOrLinkToFlag(logItem)}</>)
+    }
+    if (logItem.activity == 'deleted') {
+        descriptions.push(<>deleted the flag: {logItem.detail.name}</>)
+    }
+    if (logItem.activity == 'updated') {
+        ;(logItem.detail.changes || []).forEach((change) => {
```
Comment (mariusandra) [inline frontend/src/scenes/feature-flags/activityDescriptions.tsx:95]: nit: `;` not needed here, or swap to a `for`

**Review by mariusandra — CHANGES_REQUESTED** [review]
I think what Michael says makes sense. Let's remove [the blue bar from here](https://61260360a8e859003a7c138b-ysrpmjpcuy.chromatic.com/?path=/docs/components-activitylog--timeout#timeout) (just wanted to use storybook in an url)

You could tweak the skeletonLog to look closer to the data as well, though fine for me to go as is.

Then, can you also add tests for the `afterMount` code?

File: frontend/src/scenes/feature-flags/FeatureFlags.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/feature-flags/FeatureFlags.tsx:213]: Oh, you two always making me add nulls everywhere 🤣  #BillionDollarMistake :)

Comment (pauldambra) [reply frontend/src/scenes/feature-flags/FeatureFlags.tsx:213]: Ah, you can't pass null here

Screenshot 2022-03-21 at 12 24 18

This gets removed when we remove the feature flag

Comment (mariusandra) [reply frontend/src/scenes/feature-flags/FeatureFlags.tsx:213]: You can pass `false` or `undefined` though... so plenty of negativity to go around 😅

Comment (pauldambra) [reply frontend/src/scenes/feature-flags/FeatureFlags.tsx]: Will split the difference and pass false :)

**Review by mariusandra — APPROVED** [review]
Looks good, yet I'd still address @Twixes 's point about removing the blue "loading" bar [see here](https://61260360a8e859003a7c138b-ysrpmjpcuy.chromatic.com/?path=/docs/components-activitylog--timeout#timeout). I think it distracts a bit on the loading screen -> the blue flash makes you focus somewhere where you don't need to focus. Other than that, LGTM

**Comment by pauldambra** [comment]
> Looks good, yet I'd still address @Twixes 's point about removing the blue "loading" bar [see here](https://61260360a8e859003a7c138b-ysrpmjpcuy.chromatic.com/?path=/docs/components-activitylog--timeout#timeout). I think it distracts a bit on the loading screen -> the blue flash makes you focus somewhere where you don't need to focus. Other than that, LGTM

Yep, it's already gone. And it removed duplication in the removal 🙌

**Comment by pauldambra** [comment]
@clarkus 

I've merged this but am still working on the parent issue. So, happy to take any notes!

It's pretty close to the design, just I've called it activity instead of history

**Comment by mariusandra** [comment]
Ah, my apologies, I must have locally checked an old branch 🙈 . 

Can you also enable the flag for the team on cloud?

**Pull request merged**